clippy_lints/src/methods.rs

```diff
@@ -818,6 +851,38 @@ fn lint_iter_nth(cx: &LateContext, expr: &hir::Expr, iter_args: &MethodArgs, is_
     );
 }
 
+fn lint_get_unwrap(cx: &LateContext, expr: &hir::Expr, get_args: &MethodArgs, is_mut: bool) {
+    let mut_str = if is_mut { "_mut" } else {""};
+    let caller_type = if derefs_to_slice(cx, &get_args[0], cx.tcx.expr_ty(&get_args[0])).is_some() {
+        "slice"
+    } else if match_type(cx, cx.tcx.expr_ty(&get_args[0]), &paths::VEC) {
+        "Vec"
+    } else if match_type(cx, cx.tcx.expr_ty(&get_args[0]), &paths::VEC_DEQUE) {
+        "VecDeque"
+    } else if match_type(cx, cx.tcx.expr_ty(&get_args[0]), &paths::HASHMAP) {
+        "HashMap"
+    } else if match_type(cx, cx.tcx.expr_ty(&get_args[0]), &paths::BTREEMAP) {
+        "BTreeMap"
+    } else {
+        return; // caller is not a type that we want to lint
+    };
+
+    span_lint_and_then(
+        cx,
+        GET_UNWRAP,
+        expr.span,
+        &format!("called `.get{0}().unwrap()` on a {1}. Using `[]` is more clear and more concise",
+                 mut_str, caller_type),
+        |db| {
+            db.span_suggestion(
+                expr.span,
+                "try this",
+                format!("{}[{}]", snippet(cx, get_args[0].span, "_"), snippet(cx, get_args[1].span, "_"))
```

Comment:
As noted in the general comment, an ampersand should be added at the beginning of the suggestion string